### DISCUSSION TITLE
Compile the model online in EP

### DISCUSSION THREAD

**Discussion by wujiangGitHub** [microsoft/onnxruntime-genai#1111] [Support for Targets (OS / EPs / Hardware)]
Hello, some of the backends of onnxruntime require online compilation of models, such as openvino. Some CPUs with poor online compilation of large models may take more than 30 minutes. Is there any solution for this? Or are there any requirements for supporting such backends?

**Comment by RyanUnderhill** [comment]
Is this an onnxruntime specific issue? Or is there a solution with onnxruntime that genai is not making accessible?

**Reply by wujiangGitHub** [reply]
Thank you for your answer. 
    Because cuda and cpu are different, openvino is the backend engine that needs to compile the model online. The online compilation time of large models may be longer. I would like to ask, if jenAI supports EPs such as openvino and rknpu in the future, can the inference performance of large models meet the requirements?

**Reply by RyanUnderhill** [reply]
Ah yes, we do plan on supporting OpenVINO and compile based EPs. We will ensure the user experience is usable, as a long compilation time would not be usable. It will probably be through a separate optimized model for these providers.

**Reply by wujiangGitHub** [reply]
Thank you very much for your answer. Are these optimized models compiled and saved offline? Or can the backend provider optimize the model to improve the online compilation time to make it usable?

**Comment by ankitm3k** [comment]
Update: ORT GenAI now supports OpenVINO EP, please use the latest commits from https://github.com/microsoft/onnxruntime main branch

**Comment by wujiangGitHub** [comment]
Hello, how is the performance of the large language generation model?



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/onnxruntime-genai"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2025年7月5日(星期六) 下午3:54
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/onnxruntime-genai] Compile the model online in EP (Discussion #1111)





 
Update: ORT GenAI now supports OpenVINO EP, please use the latest commits from https://github.com/microsoft/onnxruntime main branch
